src/components/KTableData/KTableData.vue

```diff
@@ -275,39 +272,43 @@ const tablePaginationAttributes = computed((): TablePaginationAttributes => ({
   currentPage: page.value,
   offsetPreviousButtonDisabled: !previousOffset.value,
   offsetNextButtonDisabled: !nextOffset.value || !hasNextPage.value,
-  ...props.paginationAttributes,
+  ...paginationAttributes,
 }))
 
-function isTableColumnSlotName(slot: string): slot is TableColumnSlotName {
+function isTableColumnSlotName(slot: string): slot is TableColumnSlotName<ColumnKey> {
   return slot.startsWith('column-')
 }
 
-function isTableColumnTooltipSlotName(slot: string): slot is TableColumnTooltipSlotName {
+function isTableColumnTooltipSlotName(slot: string): slot is TableColumnTooltipSlotName<ColumnKey> {
   return slot.startsWith('tooltip-')
 }
 
-const getHeaderSlots = computed((): TableColumnSlotName[] => {
+function isTableColumnKey(slot: string): slot is ColumnKey {
+  return tableHeaders.value.some((header) => header.key === slot)
+}
+
+const getHeaderSlots = computed(() => {
   if (!slots) {
     return []
   }
 
   return Object.keys(slots).filter(isTableColumnSlotName)
 })
 
-const getHeaderTooltipSlots = computed((): TableColumnTooltipSlotName[] => {
+const getHeaderTooltipSlots = computed(() => {
   if (!slots) {
     return []
   }
 
   return Object.keys(slots).filter(isTableColumnTooltipSlotName)
 })
 
-const getCellSlots = computed((): string[] => {
+const getCellSlots = computed(() => {
```

Comment:
If we don't define a more explicit return type here for these computed functions I worry that over time with future edits they may change unexpectedly. Can we add return types?